xds/src/test/java/io/grpc/xds/XdsClientImplTest.java

```diff
@@ -3741,10 +3845,17 @@ private DiscoveryRequestMatcher(String versionInfo, String resourceName, String
 
     private DiscoveryRequestMatcher(String versionInfo, List<String> resourceNames, String typeUrl,
         String responseNonce) {
+      this(versionInfo, resourceNames, typeUrl, responseNonce, null);
+    }
+
+    private DiscoveryRequestMatcher(
+        String versionInfo, List<String> resourceNames, String typeUrl, String responseNonce,
+        @Nullable String nackErrorDetail) {
       this.versionInfo = versionInfo;
       this.resourceNames = new HashSet<>(resourceNames);
       this.typeUrl = typeUrl;
       this.responseNonce = responseNonce;
+      this.nackErrorDetail = nackErrorDetail;
```

Comment:
I would not add error_detail to the matcher and make it nullable is even worse. If you require checking error_detail for the message, you should check its value for all NACKs, not just test cases you just added. Why some test cases require checking error_detail in NACKs while others do not?  I would not verify error messages literally to make unit test fragile, unless the message is required to literally match some specific string.

Reply:
> Why some test cases require checking error_detail in NACKs while others do not?

I tried to check error detail in all NACK cases in this class. Not sure if I missed any.

>I would not verify error messages literally to make unit test fragile

I agree it's not very good to test literal match. But if a test is broken, it can easily be fixed. This helps find bugs than not checking error_detail, just like this one https://github.com/grpc/grpc-java/pull/6961/files#r413188115

Reply:
I see. The number of NACKs are much smaller than I expected. 

It's verbose (this test class is already extremely verbose), and verifying all NACK cases is never likely to happen given how many branches involved in XdsClientImpl's implementation. 
Also, the nullable setting of `nackErrorDetail` doesn't make tests more robust as setting it to null (or use the other constructor) will also make the test pass.

IMO, it doesn't add much value, and it adds noise to the test.

Reply:
Alright. I reverted the checking error_detail part. But to make sure the Nack is really caused by a specific error, not an Nack introduced by a bug in test code, I added a comparison test.